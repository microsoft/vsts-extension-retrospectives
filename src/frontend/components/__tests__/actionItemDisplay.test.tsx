import * as React from "react";
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import ActionItemDisplay, {
  ActionItemDisplayProps,
} from "../actionItemDisplay";

const mockOnUpdateActionItem = jest.fn(() => {});

const defaultTestProps: ActionItemDisplayProps = {
  feedbackItemId: "101",
  feedbackItemTitle: "Test Feedback Item Title",
  team: undefined,
  boardId: "Test Board Id",
  boardTitle: "Test Board Title",
  defaultIteration: "1",
  defaultAreaPath: "/testPath",
  actionItems: [],
  nonHiddenWorkItemTypes: [],
  allWorkItemTypes: [],
  allowAddNewActionItem: false,
  onUpdateActionItem: mockOnUpdateActionItem,
};

describe("Action Item Display component", () => {
  it("renders correctly when there are no action items.", () => {
    const wrapper = shallow(<ActionItemDisplay {...defaultTestProps} />);
    const component = wrapper.children().dive();
    expect(toJson(component)).toMatchSnapshot();
  });

  it("renders correctly when action items exist", () => {});
});
