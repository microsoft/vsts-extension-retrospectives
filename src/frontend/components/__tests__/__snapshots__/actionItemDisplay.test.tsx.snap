// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Item Display component renders correctly when there are no action items. 1`] = `
<div
  className="action-items"
>
  <Dialog
    dialogContentProps={
      Object {
        "title": "Link existing work item",
        "type": 0,
      }
    }
    hidden={true}
    modalProps={
      Object {
        "className": "retrospectives-dialog-modal",
        "containerClassName": "retrospectives-link-existing-work-item-dialog",
        "isBlocking": true,
      }
    }
    onDismiss={[Function]}
  >
    <StyledSearchBoxBase
      aria-label="Enter the exact work item id"
      autoFocus={true}
      className="work-item-id-input"
      onChange={[Function]}
      placeholder="Enter the exact work item id"
    />
    <div
      className="error-container"
    />
    <div
      className="output-container"
    />
    <StyledDialogFooterBase>
      <CustomizedPrimaryButton
        disabled={true}
        onClick={[Function]}
        text="Link work item"
      />
      <CustomizedDefaultButton
        onClick={[Function]}
        text="Cancel"
      />
    </StyledDialogFooterBase>
  </Dialog>
</div>
`;
